### PR TITLE
Control: show more telemetry data

### PR DESCRIPTION
**Pull request by BlackYoup**

### CODE REVIEW

File: src/software/control/Cargo.toml [hunk 1]
```diff
@@ -20,12 +20,11 @@ doc = false
...
 lazy_static = "1.4"
-chrono = "0.4"
+chrono = "0.4.11"
 winit = "0.19"
-image = "0.23"
+image = "0.23.3"
 glium = "0.24"
-plotters = {version = "0.2", default_features = false, features = ["image_encoder", "chrono", "line_series", "palette", "ttf"]}
-piston_window = "0.105"
+plotters = {version = "^0.2.12", default_features = false, features = ["image_encoder", "chrono", "line_series", "palette", "ttf"]}
```
Comment (dsferruzza) [inline src/software/control/Cargo.toml:27]: Out of curiosity: what is the difference between `0.2` and `^0.2.12` here?

File: src/software/control/src/display/drawer.rs [hunk 2]
```diff
@@ -184,49 +180,39 @@ impl DisplayDrawer {
...
-        data.insert(0, (Local::now(), new_point / 10));
-        let oldest = data.first().unwrap().0 - Duration::seconds(40);
-        let newest = data.first().unwrap().0;
-        let mut i = 0;
-        while i != data.len() {
-            if oldest > (&mut data[i]).0 || newest < (&mut data[i]).0 {
-                let _ = data.remove(i);
-            } else {
-                i += 1;
+        let now = Local::now();
```
Comment (dsferruzza) [inline src/software/control/src/display/drawer.rs:183]: Maybe we should use the `systick` field of `DataSnapshot` which is a timestamp (0 = when the MCU booted). _This can be done in a further PR._

File: src/software/telemetry/src/structures.rs [hunk 3]
```diff
@@ -55,20 +71,7 @@ pub enum TelemetryMessage {
...
-        plateau_command: u8,
-        peep_command: u8,
-        cpm_command: u8,
-        previous_peak_pressure: u8,
-        previous_plateau_pressure: u8,
-        previous_peep_pressure: u8,
-        current_alarm_codes: Vec<u8>,
-        previous_alarm_codes: Vec<u8>,
-    },
+    MachineStateSnapshot(MachineStateSnapshot),
```
Comment (dsferruzza) [inline src/software/telemetry/src/structures.rs:74]: Maybe I should do this change first in #193 so it is consistent and then you rebase on it when it is merged?

**Comment by dsferruzza** [comment]
Also I believe this could benefit from a run of `cargo fmt && cargo clippy` 🙂

File: src/software/control/src/display/drawer.rs (hunk 2 above)
Comment (BlackYoup) [reply src/software/control/src/display/drawer.rs:183]: Oh indeed, didn't know there was a timestamp sent. Will do

File: src/software/control/Cargo.toml (hunk 1 above)
Comment (BlackYoup) [reply src/software/control/Cargo.toml:27]: None, it's the same, I left it after some debug, I'll clean it a bit

File: src/software/telemetry/src/structures.rs (hunk 3 above)
Comment (BlackYoup) [reply src/software/telemetry/src/structures.rs:74]: Yep, we could do that

Comment (dsferruzza) [reply src/software/telemetry/src/structures.rs:74]: Cool! I added this change in #193

**Comment by BlackYoup** [comment]
Done for cargo fmt & clippy :+1:

File: src/software/telemetry/src/structures.rs (hunk 3 above)
Comment (BlackYoup) [reply src/software/telemetry/src/structures.rs:74]: alright, waiting for it to be merged so I can rebase

Comment (dsferruzza) [reply src/software/telemetry/src/structures.rs:74]: It's merged

**Comment by BlackYoup** [comment]
@dsferruzza I rebased on your code. I needed to add a Default derive for a struct of the telemetry crate, check out https://github.com/makers-for-life/makair/pull/194/commits/b05a5e95bbefd9702426207225a4435952e56cae

It's needed here: https://github.com/makers-for-life/makair/pull/194/files#diff-45ca8824b2a6d93856e623f283047f27R82

Let me know if that works for you

**Comment by ronanM** [comment]
@BlackYoup Can you confirm that Control can run only connected with a real MakAir device ?

**Comment by dsferruzza** [comment]
@ronanM I made few patches and I will be possible to use it with per-recorded telemetry events very soon!

**Comment by BlackYoup** [comment]
@ronanM You can now use some recorded files by using `cargo run -- -f ../telemetry/records/<pick a file>`. We will provide real records files soon, for now those are very simple and data isn't from a working makair machine.

**Pull request merged**